stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -397,6 +401,12 @@ impl NakamotoBlockBuilder {
         burn_dbconn: &'a SortitionHandleConn,
         info: &'b mut MinerTenureInfo<'a>,
     ) -> Result<ClarityTx<'b, 'b>, Error> {
+        if info.tenure_block_commit_opt.is_none() {
```

Comment:
I think it would be cleaner if you did something like this:
```rust
        let Some(tenure_block_commit) = info.tenure_block_commit_opt.is_none() else {
            return Err(Error::InvalidStacksBlock(
                "Block-commit is required; cannot mine a shadow block".into(),
            ));
        }
 ```
 and then got rid of the extra unwrap below.
 Unless of some borrowing specific reason you did it this way.